### PR TITLE
FLAC and OGG/vorbis support for audio

### PR DESCRIPTION
**Pull request by IntelOrca**
Add optional support for streaming FLAC and OGG/vorbis audio.

Galaxy style from OpenMusic is over 40 MiB in wav. FLAC gets it down to about 27 MiB and OGG/vorbis can get it down to just 1.36 MiB. I think vorbis should be our primary format for distributing OpenRCT2 assets due to its low size, but FLAC is available for those who want higher quality assets.

Depends on #17188 and requires update to dependencies.

### CODE REVIEW

**Comment by karst** [comment]
I am currently testing out this PR. 

Here are my observations:
I made an object using the OGG format: [Download](https://cdn.discordapp.com/attachments/694586451500859469/978446045384155146/rct3.music.summerair.parkobj)  
Thisone loads as intended.
My otherones using WAV ([Download](https://cdn.discordapp.com/attachments/694586451500859469/976113292521123970/openrct2.music.acid.parkobj)) also still work as intended.  

Bugs I have found:
- Title screen doesn't have any music
- When going back to the ride using the OGG tune from the WAV ride the game crashes.
image

**Comment by IntelOrca** [comment]
@karst in task manager, you can right click any process and click create dump file. Do this next time it ever hangs, and then zip it.

**Comment by karst** [comment]
Here is a dump of a hang. After a bit of playing around I got it to hang again.

The file is massive though so I had to upload it to drive: [Download](https://drive.google.com/file/d/1s3KWxY4uTcW8AtyKRTPlI8HoPoJtjdWM/view?usp=sharing)

**Comment by karst** [comment]
Ah the RCT1 title is not working since last PR was merged. I will make an issue specifically for that then, it's not the fault of this PR.

**Comment by gerion0** [comment]
Not involved at all and just curious: As far as I know, Vorbis is mostly superseeded by [Opus](https://opus-codec.org/) (fits also in OGG, is open and royalty free and much better compression ratio). Why did you choose Vorbis and not Opus?

**Comment by IntelOrca** [comment]
> Not involved at all and just curious: As far as I know, Vorbis is mostly superseeded by [Opus](https://opus-codec.org/) (fits also in OGG, is open and royalty free and much better compression ratio). Why did you choose Vorbis and not Opus?

Vorbis is ogg compatible and also open and royalty free, so that that just leaves compression ratio. Opus was giving worse compression ratios with the music from open-music. I didn't spend too long tweaking the settings, but I wasn't impressed - perhaps that was just an issue with the ffmpeg encoder. It might be because it doesn't support lower sample rates, unlike vorbis.

**Comment by karst** [comment]
SoundCloud recently switched to 64 opus, and honestly i don't think anyone with a bit of respect for audio should be using this compression. The highend starts to hiss and it's just not it. That's my personal experience with opus, though.

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
     "version": "0.2.0",
     "configurations": [
-        
         {
             "name": "C++ Launch",
             "type": "cppdbg",
             "request": "launch",
             "program": "${workspaceFolder}/bin/openrct2",
-            "args": [],
+            "args": [
```
Comment (Gymnasiast) [inline .vscode/launch.json:9]: Looks like an accidental commit.

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (Gymnasiast) [inline .vscode/tasks.json:1]: Looks like an accidental commit. (If it isn’t, putting it in a separate PR might be better.)

File: src/openrct2-android/app/src/main/CMakeLists.txt [hunk 3]
```diff
@@ -131,7 +131,7 @@ include_directories(SYSTEM "${CMAKE_BINARY_DIR}/libs/include/freetype2")
 include_directories(SYSTEM "${CMAKE_BINARY_DIR}/libs/include/SDL2")
 
 # now build app's shared lib
-add_definitions(-DDISABLE_HTTP -DDISABLE_DISCORD_RPC -DDISABLE_HTTP -DDISABLE_NETWORK -DDISABLE_OPENGL -DGL_GLEXT_PROTOTYPES -D__STDC_LIMIT_MACROS -DNO_TTF -DSDL_MAIN_HANDLED)
+add_definitions(-DDISABLE_HTTP -DDISABLE_DISCORD_RPC -DDISABLE_HTTP -DDISABLE_NETWORK -DDISABLE_FLAC -DDISABLE_VORBIS -DDISABLE_OPENGL -DGL_GLEXT_PROTOTYPES -D__STDC_LIMIT_MACROS -DNO_TTF -DSDL_MAIN_HANDLED)
```
Comment (Gymnasiast) [inline src/openrct2-android/app/src/main/CMakeLists.txt:134]: Do we want to disable this on Android? (I don’t know how difficult it is to add this to Android, perhaps @janisozaur can tell?)

File: src/openrct2/management/Award.h [hunk 4]
```diff
@@ -38,8 +38,8 @@ static_assert(static_cast<uint32_t>(AwardType::Count) < 32); // Awards are used
 
 struct Award
 {
-    uint16_t Time;
-    AwardType Type;
+    uint16_t Time{};
```
Comment (Gymnasiast) [inline src/openrct2/management/Award.h:41]: Unrelated to this PR

**Review by Gymnasiast — COMMENTED** [review]
Done with the code review, I will now move on to testing.

**Comment by gerion0** [comment]
> > Not involved at all and just curious: As far as I know, Vorbis is mostly superseeded by [Opus](https://opus-codec.org/) (fits also in OGG, is open and royalty free and much better compression ratio). Why did you choose Vorbis and not Opus?
> 
> Vorbis is ogg compatible and also open and royalty free, so that that just leaves compression ratio.

Yeah, both are from the Xiph (at least partly).

> Opus was giving worse compression ratios with the music from open-music. I didn't spend too long tweaking the settings, but I wasn't impressed - perhaps that was just an issue with the ffmpeg encoder. It might be because it wasn't taking advantage of the low sample rate compared with the vorbis encoder.

Ok, interesting to hear, thanks for clarification.

**Review by Gymnasiast — COMMENTED** [review]
Seems to work correctly, nice job! :+1:

The line notes should be easy to address.

File: src/openrct2-android/app/src/main/CMakeLists.txt (hunk 3 above)
Comment (IntelOrca) [reply src/openrct2-android/app/src/main/CMakeLists.txt:134]: Trying to add dependencies is very convoluted given all the different platforms you need to account for. I don't know how the dependencies on Android work, and it looks like it disables many of the other dependencies we also use. I didn't want Android to hold back this PR.

Comment (IntelOrca) [reply src/openrct2-android/app/src/main/CMakeLists.txt:134]: #17349

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**